Vegetarian-Food-Finder/Model/TimeZoneConverter.swift

```diff
@@ -0,0 +1,180 @@
+//
+//  TimeZoneCalculator.swift
+//  Vegetarian-Food-Finder
+//
+//  Created by Abdullahi Ahmed on 7/28/22.
+//
+
+import CoreLocation
+import Foundation
+import UIKit
+
+/// A class that is responsible for date manuplation so the user can make events at their desiered time
+final class TimeZoneConverter {
+    let selectedRestaurant: BusinessDetails
+    init(
+        selectedRestaurant: BusinessDetails
+    ) {
+        self.selectedRestaurant = selectedRestaurant
+    }
+
+    /// computed property that represents the TimeZone of the user
+    private var userCurrnetTime: TimeZone {
+        return TimeZone.current
+    }
+
+    /// Enum that handles the case if the restaurant is closed
+    enum PickedDateError: Error {
+        case invalidDate
+    }
+
+    /**
+      gets the restaurants TimeZone by using reverseGeocoding for locations
+
+     - parameter  completion handler: Gives back the TimeZone of the restaurant
+
+     - Returns: TimeZone of the restaurant
+     */
+    func restaurantLocalTime(
```

Comment:
The function name is confusing. What is doing is getting the timeZone of a restaurant